src/main/java/com/hubspot/jinjava/lib/filter/AbstractFilter.java

```diff
@@ -0,0 +1,225 @@
+/**********************************************************************
+ Copyright (c) 2014 HubSpot Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ **********************************************************************/
+package com.hubspot.jinjava.lib.filter;
+
+import com.hubspot.jinjava.doc.annotations.JinjavaDoc;
+import com.hubspot.jinjava.doc.annotations.JinjavaParam;
+import com.hubspot.jinjava.interpret.InvalidInputException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+/***
+ * Filter base that uses Filter Jinjavadoc to construct named argument parameters.
+ * Only filters that specify name, type and defaults correctly should use this as a base
+ *
+ * @see JinjavaDoc
+ * @see JinjavaParam
+ */
+public abstract class AbstractFilter implements Filter {
+  private final Map<String, JinjavaParam> NAMED_ARGUMENTS;
+
+  public AbstractFilter() {
+    NAMED_ARGUMENTS = initNamedArguments();
+  }
+
+  abstract Object filter(
+    Object var,
+    JinjavaInterpreter interpreter,
+    Map<String, Object> parsedArgs
+  );
+
+  public Object filter(Object var, JinjavaInterpreter interpreter, String... args) {
+    throw new NotImplementedException("Not implemented");
+  }
+
+  public Object filter(
+    Object var,
+    JinjavaInterpreter interpreter,
+    Object[] args,
+    Map<String, Object> kwargs
+  ) {
+    Map<String, Object> namedArgs = new HashMap<>();
+    //Set defaults
+    NAMED_ARGUMENTS.forEach(
+      (k, v) -> {
+        if (StringUtils.isNotEmpty(v.defaultValue())) {
+          namedArgs.put(k, v.defaultValue());
+        }
+      }
+    );
```

Comment:
Might as well do this in the initializer with a `stream()` and `collect()`.

Reply:
Done